pamelalozano/src/main/webapp/script.js

```diff
@@ -12,24 +12,71 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
-/**
- * Adds a random greeting to the page.
- */
-function addRandomGreeting() {
-  const greetings =
-      ['Hello world!', '¡Hola Mundo!', '你好，世界！', 'Bonjour le monde!'];
-
-  // Pick a random greeting.
-  const greeting = greetings[Math.floor(Math.random() * greetings.length)];
-
-  // Add it to the page.
-  const greetingContainer = document.getElementById('greeting-container');
-  greetingContainer.innerText = greeting;
-}
-
+/** Toggles the project's section */
 document.getElementById("projects").addEventListener("click", function (event) {
     if(event.target.classList[0]=="navbar"){
     event.target.closest('.project').classList.toggle("open");
     location.href = "#c-"+event.target.closest('.project').id;
     }
 });
+
+/** Loads existing Comments */
+async function loadComments() {
+  let response = await fetch('/data');
+  let comments = await response.json();
+  if(comments.length==0) {
+      document.getElementById("comments-section").innerHTML="No comments";
+  } 
+  else {
+    let commentSection = document.getElementById("comments-section");
+    
+    //The newest comments at the top
+    for(let i=comments.length-1; 0<=i; i--){
```

Comment:
You can also do this by reversing the comments in the list first and then using let comment of comments. This would actually change the list, though. Might be okay to do that though!

Reply:
Oh okay let me see how can I sort the json in the server side, should I insert the new comments at the beginning at the json instead of at the end (like an unshift but in java)? or use the parameterizing method?

Reply:
Yes you could insert a new comment at the front of the list. Or you could reverse the list inside `doGet()` and respond with that.

Here's a more complicated way

1. Store comments as a `ArrayList<Comment>` instead of `ArrayList<String>` i.e. don't store them in their JSON format.
2. When `doGet()` is called, you'll need to sort the comments array using `Arrays.sort(...)` and compare the dates using a custom `Comparator<Comment>`, or implement the `Comparable` interface on `Comment` to compare the date field.

Imagine you also have a field called `upvotes` on `Comment`. You could change the sort comparator to compare `upvotes` instead and show the highest upvoted comments first.

Reply:
Ohh okay okay I think I'll do this as a feature later on, when I already have the datastore. Thank you!

Reply:
Sounds good. The datastore client lets you specify things like sort-by and limit on queries, so you wouldn't have to manually sort like this.